buildspec.yml

```diff
@@ -24,17 +24,41 @@ phases:
 
       # run notebook test
       - |
-        if has-matching-changes "src/*.py" "setup.py" "setup.cfg" "buildspec.yml"; then
+        if has-matching-changes "src/*.py" "setup.py" "setup.cfg"; then
           echo "running notebook test"
           ./tests/scripts/run-notebook-test.sh
          else
            echo "skipping notebook test"
          fi
 
+      # wait its turn to run integration tests
+      - ACCOUNT=$(aws sts get-caller-identity --output text | awk '{print $1}')
+      - S3_BUCKET_DIR=s3://sagemaker-us-west-2-${ACCOUNT}/ci-lock
+
+      - |
+        while true; do
+            LAST_LOCK=$(aws s3 ls ${S3_BUCKET_DIR} | sort | tail -n 1 | awk '{print $4}')
+
+            if test -z "$LAST_LOCK"
+            then
+                echo "No locks. Creating lock."
+                LOCK_FILE=$(date +%s)
+                touch ${LOCK_FILE}
+
+                aws s3 cp ${LOCK_FILE} ${S3_BUCKET_DIR}/${LOCK_FILE}
+
+            else
```

Comment:
As per a locking system that was implemented in the Shopbop order system. You'll also want to ignore a file that is more than x hours old. To ensure dead processes don't lock forever.

Reply:
The process won't lock forever because the code build has a time limit as well. Let's include this as a future improvement.

Reply:
My concern is that a CodeBuild process dies or times out and leaves a file in S3.
In that situation, manual intervention is required to allow any future builds to run.